src/Admin/AbstractAdmin.php

```diff
@@ -994,16 +998,14 @@ final public function getDatagrid(): DatagridInterface
         return $this->datagrid;
     }
 
-    /**
-     * @phpstan-param AdminInterface<object>|null $childAdmin
-     */
     final public function getSideMenu(string $action, ?AdminInterface $childAdmin = null): ItemInterface
     {
         if ($this->isChild()) {
             return $this->getParent()->getSideMenu($action, $this);
         }
 
         $this->buildTabMenu($action, $childAdmin);
+        \assert(null !== $this->menu);
```

Comment:
what about `buildTabMenu` returning the menu?

Reply:
It follow the others `buildList`, `buildForm` etc strategy.

For instance 
```
private function buildShow(): void
    {
        if ($this->loaded['show']) {
            return;
        }

        $this->loaded['show'] = true;

        $this->show = $this->getShowBuilder()->getBaseList();
        $mapper = new ShowMapper($this->getShowBuilder(), $this->show, $this);

        $this->configureShowFields($mapper);

        foreach ($this->getExtensions() as $extension) {
            $extension->configureShowFields($mapper);
        }
    }
```

As you can see, it start with 
```
if ($this->loaded['show']) {
    return;
}
```
If we want to return the menu, 
```
if ($this->loaded['show']) {
    return $this->show;
}
```
create a phpstan issue because `this->menu` may be null

But changing this to 
```
if (null !== $this->show) {
    return $this->show;
}
```
end up an infinite loop see https://github.com/sonata-project/SonataAdminBundle/issues/6142#issuecomment-644126345

I don't want to start a big refacto about this behavior.
But maybe I can
- Return the menu (or null if the menu is already loading)
- Throwing an exception here if the buildMenu is returning null.

Reply:
Done @franmomu